integration-tests/common-test-code/src/test/java/com/vaadin/flow/quarkus/it/InstantiatorTestIT.java

```diff
@@ -0,0 +1,52 @@
+package com.vaadin.flow.quarkus.it;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.vaadin.flow.test.AbstractChromeIT;
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+
+@QuarkusIntegrationTest
+public class InstantiatorTestIT extends AbstractChromeIT {
+
+    @Test
+    public void navigationCorrectlyHandlesProxiedViews() {
+        open();
+
+        String prevUuid = null;
+        AtomicReference<String> prevCounter = new AtomicReference<>("");
+        for (int i = 0; i < 5; i++) {
+            String uuid = waitUntil(d -> d.findElement(By.id("COMPONENT_ID")))
+                    .getText();
+
+            waitUntil(d -> !prevCounter.get()
+                    .equals(d.findElement(By.id("CLICK_COUNTER")).getText()));
+
+            if (prevUuid != null) {
+                Assert.assertEquals("UUID should not have been changed",
+                        prevUuid, uuid);
+            }
```

Comment:
As the test actually tests that the initial UUID stays through all navigations, should we instead only initialize the uuid once?
```suggestion
            if (prevUuid != null) {
                Assert.assertEquals("UUID should not have been changed",
                        prevUuid, uuid);
            } else {
                prevUuid = uuid;
            }
```
perhaps rename prevUuid to viewUuid or something more static.

Reply:
Good catch! I'll fix it also on Flow test

Reply:
done